### PR TITLE
DRY up logic and UI for defining form fields

### PR DESCRIPTION
**Pull request by jits**
For reuse in an upcoming feature we're currently working on.

### CODE REVIEW

File: platform-hub-api/app/models/concerns/validate_hashes.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+      validate :validate_hashes_validation
+
+      # Check that config provided is valid
+      config_ok = config.is_a?(Hash) &&
+        config.keys.any? &&
+        config.values.all? { |h| (h.keys - CONFIG_ENTRY_ALLOWED_KEYS).empty? }
+
+      unless config_ok
+        raise 'Invalid config provided to validate_hashes'
```
Comment (marcinc) [inline platform-hub-api/app/models/concerns/validate_hashes.rb]: Maybe we should throw specific exceptions with message?

File: platform-hub-api/app/services/form_fields_service.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  def all_types
+    ALL_ALLOWED_TYPES
+  end
+
+  def validate_types types
+    if types.proper_subset?(ALL_ALLOWED_TYPES)
+      types
+    else
+      raise('Invalid form field type(s) specified')
```
Comment (marcinc) [inline platform-hub-api/app/services/form_fields_service.rb]: Probably worth throwing specific error here too?

File: platform-hub-api/app/models/concerns/validate_hashes.rb (hunk 1 above)
Comment (jits) [inline platform-hub-api/app/models/concerns/validate_hashes.rb]: Given this only runs at initial load time (i.e. when models are first loaded in memory), I figured it didn't need a specific exception, especially as it's a halting error (i.e. stops the server from loading at all).

File: platform-hub-api/app/services/form_fields_service.rb (hunk 2 above)
Comment (jits) [inline platform-hub-api/app/services/form_fields_service.rb]: Same thing as above.

File: platform-hub-api/app/models/concerns/validate_hashes.rb (hunk 1 above)
Comment (marcinc) [inline platform-hub-api/app/models/concerns/validate_hashes.rb]: Makes sense.

**Comment by jits** [comment]
Cheers @marcinc !

**Pull request merged**